appdatabase/database.go

```diff
@@ -107,3 +115,71 @@ func GetDBFilename(db *sql.DB) (string, error) {
 
 	return "", errors.New("no main database found")
 }
+
+func MigrateEnsUsernames(db *sql.DB) error {
+
+	// 1. Check if ens_usernames table already exist
+
+	row := db.QueryRow("SELECT exists(SELECT name FROM sqlite_master WHERE type='table' AND name='ens_usernames')")
+	tableExists := false
+	err := row.Scan(&tableExists)
+
+	if err != nil && err != sql.ErrNoRows {
+		return err
+	}
+
+	if tableExists {
+		return nil
+	}
+
+	// -- 1. Create new ens_usernames table
+
+	_, err = db.Exec(`CREATE TABLE IF NOT EXISTS ens_usernames (
+		"username" TEXT NOT NULL,
+		"chain_id" UNSIGNED BIGINT DEFAULT 1);`)
+
+	if err != nil {
+		log.Error("Migrating ens usernames: failed to create table")
+		return err
+	}
+
+	// -- 2. Move current `settings.usernames` to the new table
+	/*
+		INSERT INTO ens_usernames (username)
+			SELECT json_each.value FROM settings, json_each(usernames);
+	*/
+
+	rows, err := db.Query(`SELECT usernames FROM settings`)
+
+	if err != nil {
+		log.Error("Migrating ens usernames: failed to query 'settings.usernames'")
+		return err
+	}
+
+	defer rows.Close()
+
+	var usernames []string
+
+	for rows.Next() {
+		var usernamesJson string
+		err := rows.Scan(&usernamesJson)
+
+		if err != nil {
+			return err
+		}
+
+		var list []string
+		json.Unmarshal([]byte(usernamesJson), &list)
+		usernames = append(usernames, list...)
+	}
+
+	for _, username := range usernames {
+		_, err := db.Exec(`INSERT INTO ens_usernames (username) VALUES (?)`, username)
+		if err != nil {
+			log.Error("Migrating ens usernames: failed to insert username into new database")
+			return err
+		}
+	}
```

Comment:
What should happen if one of the inserts fail? is it okay if the table contains incomplete results? if no, consider using a transaction and rolling back on error